lib/internal/url.js

```diff
@@ -1504,50 +1502,39 @@ function fileURLToPath(path) {
 // - CR: The carriage return character is also stripped out by the `pathname`
 //       setter.
 // - TAB: The tab character is also stripped out by the `pathname` setter.
-const percentRegEx = /%/g;
-const backslashRegEx = /\\/g;
-const newlineRegEx = /\n/g;
-const carriageReturnRegEx = /\r/g;
-const tabRegEx = /\t/g;
-
 function encodePathChars(filepath) {
-  if (StringPrototypeIncludes(filepath, '%'))
-    filepath = StringPrototypeReplace(filepath, percentRegEx, '%25');
+  filepath = StringPrototypeReplaceAll(filepath, '%', '%25');
   // In posix, backslash is a valid character in paths:
-  if (!isWindows && StringPrototypeIncludes(filepath, '\\'))
-    filepath = StringPrototypeReplace(filepath, backslashRegEx, '%5C');
-  if (StringPrototypeIncludes(filepath, '\n'))
-    filepath = StringPrototypeReplace(filepath, newlineRegEx, '%0A');
-  if (StringPrototypeIncludes(filepath, '\r'))
-    filepath = StringPrototypeReplace(filepath, carriageReturnRegEx, '%0D');
-  if (StringPrototypeIncludes(filepath, '\t'))
-    filepath = StringPrototypeReplace(filepath, tabRegEx, '%09');
+  if (!isWindows) filepath = StringPrototypeReplaceAll(filepath, '\\', '%5C');
+  filepath = StringPrototypeReplaceAll(filepath, '\n', '%0A');
+  filepath = StringPrototypeReplaceAll(filepath, '\r', '%0D');
+  filepath = StringPrototypeReplaceAll(filepath, '\t', '%09');
   return filepath;
 }
 
 function pathToFileURL(filepath) {
   const outURL = new URL('file://');
   if (isWindows && StringPrototypeStartsWith(filepath, '\\\\')) {
     // UNC path format: \\server\share\resource
-    const paths = StringPrototypeSplit(filepath, '\\');
-    if (paths.length <= 3) {
+    const hostnameEndIndex = StringPrototypeIndexOf('\\', 2);
+    if (hostnameEndIndex === -1) {
       throw new ERR_INVALID_ARG_VALUE(
         'filepath',
         filepath,
         'Missing UNC resource path'
       );
     }
-    const hostname = paths[2];
-    if (hostname.length === 0) {
+    if (hostnameEndIndex === 3) {
       throw new ERR_INVALID_ARG_VALUE(
         'filepath',
         filepath,
         'Empty UNC servername'
       );
     }
+    const hostname = StringPrototypeSlice(filepath, 2, hostnameEndIndex);
     outURL.hostname = domainToASCII(hostname);
     outURL.pathname = encodePathChars(
-      ArrayPrototypeJoin(ArrayPrototypeSlice(paths, 3), '/'));
+      StringPrototypeReplaceAll(StringPrototypeSlice(filepath, hostnameEndIndex), '\\', '/'));
```

Comment:
This pull request changes more than using refactoring primordials.

Reply:
`StringPrototypeSplit` relies on user-mutable `String.prototype[Symbol.split]`, so it's not exactly unrelated.